dpreference/src/main/java/me/dozen/dpreference/PreferenceProvider.java

```diff
@@ -1,67 +1,78 @@
 package me.dozen.dpreference;
 
-import android.content.ContentProvider;
-import android.content.ContentValues;
-import android.content.UriMatcher;
-import android.database.Cursor;
-import android.database.MatrixCursor;
-import android.net.Uri;
-import android.support.annotation.Nullable;
-import android.support.v4.util.ArrayMap;
-import android.text.TextUtils;
-import android.util.Log;
-
-import java.util.List;
+import ohos.aafwk.ability.Ability;
+import ohos.aafwk.ability.PathMatcher;
+import ohos.aafwk.content.Intent;
+import ohos.agp.utils.TextTool;
+import ohos.data.dataability.DataAbilityPredicates;
+import ohos.data.rdb.ValuesBucket;
+import ohos.data.resultset.ResultSet;
+import ohos.data.resultset.TableResultSet;
+import ohos.hiviewdfx.HiLog;
+import ohos.hiviewdfx.HiLogLabel;
+import ohos.utils.LightweightMap;
+import ohos.utils.net.Uri;
 import java.util.Map;
 
 /**
  * Created by wangyida on 15/12/18.
  */
-public class PreferenceProvider extends ContentProvider {
-
+public class PreferenceProvider extends Ability {
     private static final String TAG = PreferenceProvider.class.getSimpleName();
-
     private static final String AUTHORITY = "me.dozen.dpreference.PreferenceProvider";
-
-    public static final String CONTENT_PREF_BOOLEAN_URI = "content://" + AUTHORITY + "/boolean/";
-    public static final String CONTENT_PREF_STRING_URI = "content://" + AUTHORITY + "/string/";
-    public static final String CONTENT_PREF_INT_URI = "content://" + AUTHORITY + "/integer/";
-    public static final String CONTENT_PREF_LONG_URI = "content://" + AUTHORITY + "/long/";
-
-
+    private static final String ACTION = "values is null!!!";
+    private static final String DATA_ABILITY = "dataability:///";
+    public static final String CONTENT_PREF_BOOLEAN_URI =  DATA_ABILITY + AUTHORITY + "/boolean/";
+    public static final String CONTENT_PREF_STRING_URI =  DATA_ABILITY + AUTHORITY + "/string/";
+    public static final String CONTENT_PREF_INT_URI =  DATA_ABILITY + AUTHORITY + "/integer/";
+    public static final String CONTENT_PREF_LONG_URI =  DATA_ABILITY + AUTHORITY + "/long/";
     public static final String PREF_KEY = "key";
     public static final String PREF_VALUE = "value";
-
     public static final int PREF_BOOLEAN = 1;
     public static final int PREF_STRING = 2;
     public static final int PREF_INT = 3;
     public static final int PREF_LONG = 4;
-
-    private static final UriMatcher sUriMatcher;
+    public static final HiLogLabel HI_LOG_LABEL_2 = new HiLogLabel(0, 0, TAG);
```

Comment:
Need not be public variable. Change it to private and remove the suffix "_2" from the variable name

Reply:
Done